io/src/pcd_io.cpp

```diff
@@ -467,60 +467,31 @@ pcl::PCDReader::readBodyASCII (std::istream &fs, pcl::PCLPointCloud2 &cloud, int
         }
         for (unsigned int c = 0; c < cloud.fields[d].count; ++c)
         {
+#define COPY_STRING(CASE_LABEL)                                                          \
```

Comment:
Is there maybe a possibility to remove the `switch` completely and map directly from `cloud.fields[d].datatype` to the template parameter?

Reply:
No. The datatype is runtime, while template is compile time. 😢